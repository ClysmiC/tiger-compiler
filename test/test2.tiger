/**
 * This file mainly tests array access and storage, as well as a function call.
 * Its output isn't useful, but can easily be verified by running the program mentally.
 */

let
	type ArrayInt = array[2] of int;

	var buffer : ArrayInt;
	var y : int := 7;
	var z : int := 20;

	function fillArray( firstInt : int, secondInt: int)
        begin
        	buffer[0] := firstInt;
        	buffer[1] := secondInt;
        end;
in
	fillArray(y, z);

	/* essentially, buffer[1] *= buffer[0]. just testing complex code generation */
	buffer[1] := buffer[0 + 1 + 9 * 2 - 19] * buffer[1]; 

	printi(buffer[0]);
	printi(buffer[1]);
end