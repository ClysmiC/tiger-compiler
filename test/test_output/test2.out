LET
FUNCTION
ID
LPAREN
ID
COLON
INT
RPAREN
BEGIN
ID
LPAREN
ID
RPAREN
SEMI
END
SEMI
IN
ID
LPAREN
INTLIT
RPAREN
SEMI
END
EOF

Successful parse